packages/gestalt/src/Tooltip.js

```diff
@@ -3,62 +3,128 @@
+  const { isOpen } = state;
+
+  const childRef = React.useRef<?HTMLDivElement>(null);
+  const { current: anchor } = childRef;
+
+  const handleIconMouseEnter = () => {
+    dispatch({ type: 'hoverInIcon' });
   };
 
-  childRef: {| current: null | React.ElementRef<'div'> |} = React.createRef();
+  const handleIconMouseLeave = () => {
+    setTimeout(() => {
+      dispatch({ type: 'hoverOutIcon' });
+    }, 75);
+  };
 
-  handleMouseEnter = () => this.setState({ hovered: true });
+  const handleTextMouseEnter = () => {
+    dispatch({ type: 'hoverInText' });
+  };
 
-  handleMouseLeave = () => this.setState({ hovered: false });
+  const handleTextMouseLeave = () => {
+    setTimeout(() => {
+      dispatch({ type: 'hoverOutText' });
+    }, 75);
+  };
 
-  render() {
-    const { children, inline, text } = this.props;
-    const { hovered } = this.state;
-    const { current: anchor } = this.childRef;
+  const allowTooltipHover = href && hrefText;
 
-    return (
-      <Box display={inline ? 'inlineBlock' : 'block'}>
-        <Box
-          onMouseEnter={this.handleMouseEnter}
-          onMouseLeave={this.handleMouseLeave}
-          ref={this.childRef}
+  return (
+    <Box display={inline ? 'inlineBlock' : 'block'}>
+      <Box
+        ref={childRef}
+        onMouseEnter={handleIconMouseEnter}
+        onMouseLeave={handleIconMouseLeave}
+        onFocus={handleIconMouseEnter}
+        onBlur={handleIconMouseLeave}
+        role="tooltip-container"
+      >
+        {children}
+      </Box>
+      {isOpen && !!anchor && (
+        <Controller
+          anchor={anchor}
+          bgColor="darkGray"
+          idealDirection="down"
+          onDismiss={noop}
+          positionRelativeToAnchor
+          size={null}
         >
-          {children}
-        </Box>
-        {hovered && !!anchor && (
-          <Controller
-            anchor={anchor}
-            bgColor="darkGray"
-            idealDirection="down"
-            onDismiss={noop}
-            positionRelativeToAnchor
-            size={null}
+          <Box
+            maxWidth={180}
+            paddingY={1}
+            paddingX={2}
+            role="tooltip"
+            onBlur={allowTooltipHover ? handleTextMouseLeave : undefined}
+            onFocus={allowTooltipHover ? handleTextMouseEnter : undefined}
+            onMouseEnter={allowTooltipHover ? handleTextMouseEnter : undefined}
+            onMouseLeave={allowTooltipHover ? handleTextMouseLeave : undefined}
+            tabIndex={allowTooltipHover ? 0 : undefined}
           >
-            <Box maxWidth={180} paddingY={1} paddingX={2} role="tooltip">
-              <Text color="white" size="xs">
-                {text}
-              </Text>
-            </Box>
-          </Controller>
-        )}
-      </Box>
-    );
-  }
+            <Text color="white" size="xs">
+              {text}
+            </Text>
+            {allowTooltipHover && (
+              <Box paddingY={1} role="href-link">
```

Comment:
@anniejiwon this is an invalid ARIA role, can we please remove it? https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques#Roles

Also see the second rule of ARIA: https://www.w3.org/TR/using-aria/#secondrule `Do not change native semantics, unless you really have to.`